android/src/main/java/org/umhan35/RNSearchBarManager.java

```diff
@@ -0,0 +1,239 @@
+            case "characters":
+                flags |= InputType.TYPE_TEXT_FLAG_CAP_CHARACTERS;
+                break;
+
+            case "sentences":
+                flags |= InputType.TYPE_TEXT_FLAG_CAP_SENTENCES;
+                break;
+
+            case "words":
+                flags |= InputType.TYPE_TEXT_FLAG_CAP_WORDS;
+                break;
+
+            default:
+                break;
+        }
+
+        searchView.setInputType(flags);
+    }
+
+    @ReactProp(name = "editable")
+    public void setEditable(SearchView searchView, boolean editable) {
+        if (!editable) {
+            processViewRecursive(searchView, view -> view.setEnabled(false));
+        }
+    }
+
+    @ReactProp(name = "keyboardType")
+    public void setKeyboardType(SearchView searchView, String type) {
+        int flags = searchView.getInputType();
+
+        switch (type) {
+            case "email-address":
+                flags = InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS;
+                break;
+
+            case "numeric":
+                flags = InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_VARIATION_NORMAL;
+                break;
+
+            case "phone-pad":
+                flags = InputType.TYPE_CLASS_PHONE;
+                break;
+
+            default:
+                // Remove class and variants.
+                flags &= ~(InputType.TYPE_MASK_CLASS | InputType.TYPE_MASK_VARIATION);
+
+                flags |= InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_NORMAL;
+                break;
+        }
+
+        searchView.setInputType(flags);
+    }
+
+    @ReactProp(name = "placeholder")
+    public void setPlaceholder(SearchView searchView, @Nullable String placeholder) {
+        searchView.setQueryHint(placeholder);
+    }
+
+    @ReactProp(name = "returnKeyType")
+    public void setReturnKeyType(SearchView searchView, String type) {
+        int flags = searchView.getImeOptions();
+
+        // Remove action.
+        flags &= ~EditorInfo.IME_MASK_ACTION;
+
+        switch (type) {
+            case "done":
+                flags |= EditorInfo.IME_ACTION_DONE;
+                break;
+
+            case "go":
+                flags |= EditorInfo.IME_ACTION_GO;
+                break;
+
+            case "next":
+                flags |= EditorInfo.IME_ACTION_NEXT;
+                break;
+
+            case "send":
+                flags |= EditorInfo.IME_ACTION_SEND;
+                break;
+
+            default:
+                flags |= EditorInfo.IME_ACTION_SEARCH;
+                break;
+        }
+
+        searchView.setImeOptions(flags);
+    }
+
+    @ReactProp(name = "text")
+    public void setText(SearchView searchView, String text) {
+        searchView.setQuery(text, false);
+    }
+
+    @ReactProp(name = "textColor", customType = "Color")
+    public void setTextColor(SearchView searchView, @Nullable Integer color) {
+        if (color != null) {
```

Comment:
when someone specifies `textColor` prop and then resets it to `undefined`, then `color` here will be `null` and the JS prop change will be ignored

Reply:
Harder problem than it looks at first.  When undefined, the system default text color is used.  If we are explicitly setting it back to default, we must determine what said color was (awkward, but possible).  *However*, default system behavior changes the color of the text when it is disabled, but does *not* do so when we override via this function.

Reply:
Figured it out. Fixed now.

Reply:
Thanks @alpha0010 for your hard work and research here!!